src/content/docs/en/guides/backend/supabase.mdx

```diff
@@ -3,10 +3,809 @@ title: Supabase & Astro
+3. Add the following code to `signup.astro`:
+
+```astro title="src/pages/signup.astro"
+---
+import Layout from "../layouts/Layout.astro";
+import { getSessionData, setSessionData } from "../lib/auth"
+
+let accessToken = Astro.cookies.get("sb-access-token").value
+let refreshToken = Astro.cookies.get("sb-refresh-token").value
+
+if (accessToken && refreshToken) {
+    await setSessionData(accessToken, refreshToken)
+    return Astro.redirect("/profile")
+}
+
+console.log(await getSessionData())
+---
+
+<Layout title="Sign Up">
+    <div class="flex justify-center items-center">
+        <div class="col-6 form-widget" aria-live="polite">
+            <h1 class="text-3xl text-gray-800">Sign Up</h1>
+            <form
+                class="flex flex-col space-y-4 mt-8"
+                aria-live="polite"
+            >
+                <input
+                    class="border rounded px-3 py-2"
+                    aria-live="polite"
+                    name="email"
+                    type="email"
+                    placeholder="Email"
+                />
+                <input
+                    class="border rounded px-3 py-2"
+                    aria-live="polite"
+                    name="password"
+                    type="password"
+                    placeholder="Password"
+                />
+                <input
+                    class="border rounded px-3 py-2"
+                    aria-live="polite"
+                    name="confirm-password"
+                    type="password"
+                    placeholder="Confirm Password"
+                />
+                <button
+                    class="block px-3 py-2 w-full text-white bg-gray-800 border rounded hover:border-2"
+                    aria-live="polite"
+                    type="submit"
+                >
+                    Sign Up
+                </button>
+                <p class="text-center text-gray-800">Already have an account? <a href="/signin">Sign In</a></p>
+            </form>
+        </div>
+    </div>        
+</Layout>
+
+<script>
+    import { signUp } from "../lib/auth";
+
+    const form = document.querySelector("form")
+
+    form?.addEventListener("submit", async (e) => {
+        e.preventDefault();
+        console.log("submitted")
+        const formData = new FormData(e.target as HTMLFormElement);
+        const creds = {
+            email: formData.get("email") as string,
+            password: formData.get("password") as string,
+            confirmPassword: formData.get("confirm-password") as string,
+        };
+        
+        if (creds.password !== creds.confirmPassword) {
+            console.log("Passwords do not match");
+            alert("Passwords do not match");
+            return;
+        } else {
+            try {
+                await signUp(creds.email, creds.password);
+            } catch (error) {
+                console.log(error);
+            }
+        }
+    });
+</script>
+```
+
```

Comment:
``````suggestion
```astro
---
// src/pages/signup.astro
import Layout from "../layouts/Layout.astro";
import { getSessionData, setSessionData } from "../lib/auth"

let accessToken = Astro.cookies.get("sb-access-token").value
let refreshToken = Astro.cookies.get("sb-refresh-token").value

if (accessToken && refreshToken) {
    await setSessionData(accessToken, refreshToken)
    return Astro.redirect("/profile")
}

console.log(await getSessionData())
---

<Layout title="Sign Up">
    <div class="flex justify-center items-center">
        <div class="col-6 form-widget" aria-live="polite">
            <h1 class="text-3xl text-gray-800">Sign Up</h1>
            <form
                class="flex flex-col space-y-4 mt-8"
                aria-live="polite"
            >
                <input
                    class="border rounded px-3 py-2"
                    aria-live="polite"
                    name="email"
                    type="email"
                    placeholder="Email"
                />
                <input
                    class="border rounded px-3 py-2"
                    aria-live="polite"
                    name="password"
                    type="password"
                    placeholder="Password"
                />
                <input
                    class="border rounded px-3 py-2"
                    aria-live="polite"
                    name="confirm-password"
                    type="password"
                    placeholder="Confirm Password"
                />
                <button
                    class="block px-3 py-2 w-full text-white bg-gray-800 border rounded hover:border-2"
                    aria-live="polite"
                    type="submit"
                >
                    Sign Up
                </button>
                <p class="text-center text-gray-800">Already have an account? <a href="/signin">Sign In</a></p>
            </form>
        </div>
    </div>        
</Layout>

<script>
    import { signUp } from "../lib/auth";

    const form = document.querySelector("form")

    form?.addEventListener("submit", async (e) => {
        e.preventDefault();
        console.log("submitted")
        const formData = new FormData(e.target as HTMLFormElement);
        const creds = {
            email: formData.get("email") as string,
            password: formData.get("password") as string,
            confirmPassword: formData.get("confirm-password") as string,
        };
        
        if (creds.password !== creds.confirmPassword) {
            console.log("Passwords do not match");
            alert("Passwords do not match");
            return;
        } else {
            try {
                await signUp(creds.email, creds.password);
            } catch (error) {
                console.log(error);
            }
        }
    });
</script>
```

``````

Reply:
following up the suggestion we should also remove styles and console logs to clear up the markup.